✨ BTT Octopus Pro V1.0.1 (STM32H723ZE)

thisiskeithb opened the pull request:
### Description

Add BigTreeTech Octopus Pro V1.0.1 (STM32H723ZE) support as found in https://github.com/MarlinFirmware/Marlin/issues/26044#issuecomment-1666954952.

The original Octopus Pro 1.0 is STM32F446ZET6-based, so the updated Octopus Pro V1.0.1 is more in line with the Octopus Pro V1.1 which I added in https://github.com/MarlinFirmware/Marlin/pull/26043.

 Pin | Octopus Pro V1.0 (STM32F446ZET6) & <br /> V1.0.1 (STM32H723ZE) | Octopus Pro V1.1 (STM32H723ZE)
  -- | -- | --
  HE0 | PA2  | PA0
  HE2 | PB10  | PB0
  Motor4-EN | PA0  | PA2
  RGB | PB0  | PB10

Minor note: I had to drop "BTT" from the Octopus Pro V1.0.1 `BOARD_INFO_NAME` since it was longer than 20 characters.

### Related Issues

- https://github.com/MarlinFirmware/Marlin/issues/26044#issuecomment-1666954952
- https://github.com/MarlinFirmware/Marlin/pull/26043

thisiskeithb commented:
@Hman32: Give this a try.

Hman32 commented:
> @Hman32: Give this a try.

Hi @thisiskeithb , it compile but I've this warning:

```
Compiling .pio\build\STM32H723ZE_btt\src\src\inc\Warnings.cpp.o
In file included from Marlin\src\inc\../pins/stm32h7/pins_BTT_OCTOPUS_PRO_V1_0_1.h:44,
                 from Marlin\src\inc\../pins/pins.h:827,
                 from Marlin\src\inc\MarlinConfig.h:36,
                 from Marlin\src\inc\Warnings.cpp:28:
Marlin\src\inc\../pins/stm32h7/pins_BTT_OCTOPUS_PRO_V1_common.h:493: warning: "NEOPIXEL_PIN" redefined
  493 |         #define NEOPIXEL_PIN         EXP1_06_PIN
      |
In file included from Marlin\src\inc\../pins/pins.h:827,
                 from Marlin\src\inc\MarlinConfig.h:36,
                 from Marlin\src\inc\Warnings.cpp:28:
Marlin\src\inc\../pins/stm32h7/pins_BTT_OCTOPUS_PRO_V1_0_1.h:41: note: this is the location of the previous definition
   41 |   #define NEOPIXEL_PIN                      PB0
      |

```      
This warning come because i use a "FYSETC_MINI_12864_2_1".
Could it be that the code
```
#ifndef NEOPIXEL_PIN
  #define NEOPIXEL_PIN                      PB10
#endif

```
Should come after the 
`#include "pins_BTT_OCTOPUS_PRO_V1_common.h"`
?

thisiskeithb commented:
> Should come after the
> #include "pins_BTT_OCTOPUS_PRO_V1_common.h"
> ?

I've rebased this PR on current `bugfix-2.1.x` and moved the `pins_BTT_OCTOPUS_PRO_V1_common.h` include so you won't get redefined warnings out of the gate.

That doesn't solve the main issue though. Marlin doesn't handle multiple NeoPixels devices very well when paired with a NeoPixel-based LCD, so you will also end up with redefined warnings if you have a display that uses NeoPixels & a NeoPixel strip (or two) enabled. This problem is only exacerbated by boards that ship with more & more NeoPixel ports.

I created a Feature Request a while ago in hopes of getting this solved, but there is no ETA on when a "smarter" NeoPixel system will be implemented: https://github.com/MarlinFirmware/Marlin/issues/25236

If you have multiple NeoPixel devices enabled, it's best to manually edit the pins file to change either the LCD or general NeoPixel pin define to `NEOPIXEL2_PIN` depending on which device you want to be "first" in the Lights menu.

Hman32 commented:
Many thanks for your support @thisiskeithb 
Will sync my fork with the latest release of  bugfix-2.1.x then will plan to replace my current board with the Octopus Pro to test it on my Core XY ;)

Hman32 commented:
Hi @thisiskeithb , I've just finished to migrate my Tronxy X5sa to the Octopus Pro and was able to properly flash the firmware.
Everything work:
* I'm able to home
* I'm using a BL Touch without issue's
* I'm able to auto z align both motor
* I'm able to level the bed
* I'm able to heat the bed and the extruder
* I could initialise the eeprom and store the settings

Now I just need to prepare some print and test the new setup but looking good so far ;)

Hman32 commented:
@thisiskeithb any idea when this will be part of bugfix2.1 branch ?

thisiskeithb commented:
> any idea when this will be part of bugfix2.1 branch ?

There’s no timeline. You can use my branch / PR source and I’ll continue to keep it up to date.

thisiskeithb committed:
✨ BTT Octopus Pro V1.0.1 (STM32H723ZE)

sjasonsmith commented:
@thisiskeithb do you have any work pending on this, or does it just need to be reviewed and go in?

thisiskeithb commented:
> @thisiskeithb do you have any work pending on this, or does it just need to be reviewed and go in?

Just needs to be reviewed and merged, please 🙏

All the force-pushes are to rebase on current bugfix / keep it up to date.
